src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.hpp

```diff
@@ -0,0 +1,164 @@
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+namespace Tags {
+template <typename Tag>
+struct dt;
+}  // namespace Tags
+/// \endcond
+
+// IWYU pragma: no_include <pup.h>
+
+namespace gr {
+namespace Solutions {
+
+/*!
+ * \brief Gauge wave in flat spacetime
+ *
+ * \details
+ * This solution is Minkowski space in coordinates chosen to contain
+ * a gauge wave. The spacetime metric is given by Eq. (4.3) of
+ * \cite Alcubierre2003pc :
+ *
+ * \f{equation}{
+ * ds^2 = -H dt^2 + H dx^2 + dy^2 + dz^2,
+ * \f} where
+ * \f{equation}{
+ * H = H(x-t) = 1 - A \sin\left(\frac{2\pi(x-t)}{d}\right).
+ * \f} The gauge wave has amplitude \f$A\f$, wavelength \f$d\f$, and propagates
+ * along the x axis.
+ *
+ * In these coordinates, the spatial metric \f$\gamma_{ij}\f$ and
+ * inverse spatial metric \f$\gamma^{ij}\f$ are diagonal,
+ * with the diagonal elements equal to unity except for
+ * \f{eqnarray}{
+ * \gamma_{xx} & = & H,\\
+ * \gamma^{xx} & = & 1/H.
+ * \f} The components of the derivatives of \f$\gamma_{ij}\f$ vanish except for
+ * \f{eqnarray}{
+ * \partial_t \gamma_{xx} & = & \partial_t H = - \partial_x H,\\
+ * \partial_x \gamma_{xx} & = & \partial_x H.
+ * \f} The square root of the spatial metric determinant is
+ * \f{eqnarray}{
+ * \sqrt{\gamma} & = & \sqrt{H}.
+ * \f}
+ *
+ * The lapse and its derivatives are
+ * \f{eqnarray}{
+ * \alpha & = & \sqrt{H},\\
+ * \partial_t \alpha & = & -\frac{\partial_x H}{2\sqrt{H}},\\
+ * \partial_x \alpha & = & \frac{\partial_x H}{2\sqrt{H}},\\
+ * \partial_y \alpha & = & \partial_z \alpha = 0.
+ * \f} The shift \f$\beta^i\f$ and its derivatives vanish.
+ *
+ * The extrinsic curvature's components vanish, except for
+ * \f{eqnarray}{
+ * K_{xx} & = & \frac{\partial_x H}{2 \sqrt{H}}.
+ * \f}
+ *
+ * The following are input file options that can be specified:
+ *  - Amplitude (default: 1.)
+ *  - Wavelength (default: 1.)
+ */
+class GaugeWave : public MarkAsAnalyticSolution {
+ public:
+  static constexpr size_t volume_dim = 3;
+  struct Amplitude {
+    using type = double;
+    static constexpr OptionString help = {"Amplitude of the gauge wave"};
+    static type default_value() noexcept { return 1.; }
+    static type lower_bound() noexcept { return 0.; }
+  };
+  struct Wavelength {
+    using type = double;
+    static constexpr OptionString help = {"Wavelength of the gauge wave"};
+    static type default_value() noexcept { return 1.; }
+    static type lower_bound() noexcept { return 0.; }
+  };
+
+  using options = tmpl::list<Amplitude, Wavelength>;
+  static constexpr OptionString help{"Gauge wave in flat spacetime"};
+
+  GaugeWave(double amplitude, double wavelength,
+            const OptionContext& context = {});
+
+  explicit GaugeWave(CkMigrateMessage* /*unused*/) noexcept {}
+
+  GaugeWave() = default;
+  GaugeWave(const GaugeWave& /*rhs*/) = default;
+  GaugeWave& operator=(const GaugeWave& /*rhs*/) = default;
```

Comment:
do you need the copy constructors?

Reply:
Yes: WrappedGr tests copy semantics. I realize that I didn't add this new solution to my test of WrappedGr, though, so I'll do that in my fixup commit.